common/src/lazy_pages.rs

```diff
@@ -23,48 +23,47 @@ use gear_core::{
     ids::ProgramId,
     memory::{HostPointer, Memory, PageBuf, PageNumber},
 };
-use gear_runtime_interface::{gear_ri, GetReleasedPageError, MprotectError};
+use gear_runtime_interface::{gear_ri, RIError};
 use sp_std::{
     collections::{btree_map::BTreeMap, btree_set::BTreeSet},
     vec::Vec,
 };
 
 #[derive(Debug, Clone, PartialEq, Eq, derive_more::Display)]
 pub enum Error {
-    #[display(fmt = "{}", _0)]
-    Mprotect(MprotectError),
-    #[display(fmt = "{}", _0)]
-    GetReleasedPage(GetReleasedPageError),
-    #[display(fmt = "Cannot convert vec to page data")]
-    VecToPageData,
-    #[display(fmt = "Cannot find page data in storage")]
-    NoPageDataInStorage,
-    #[display(fmt = "Released page {:?} has initial data", _0)]
-    ReleasedPageHasData(PageNumber),
+    #[display(fmt = "RUNTIME INTERFACE ERROR: {}", _0)]
+    RIError(RIError),
+    #[display(fmt = "RUNTIME INTERFACE ERROR: {:?} has no released data", _0)]
+    ReleasedPageHasNoData(PageNumber),
+    #[display(fmt = "RUNTIME ERROR: released page {:?} has initial data", _0)]
+    ReleasedPageHasInitialData(PageNumber),
+    #[display(fmt = "RUNTIME ERROR: wasm memory buffer is undefined")]
+    WasmMemBufferIsUndefined,
 }
 
-impl From<MprotectError> for Error {
-    fn from(err: MprotectError) -> Self {
-        Self::Mprotect(err)
-    }
-}
-
-impl From<GetReleasedPageError> for Error {
-    fn from(err: GetReleasedPageError) -> Self {
-        Self::GetReleasedPage(err)
+impl From<RIError> for Error {
+    fn from(err: RIError) -> Self {
+        Self::RIError(err)
     }
 }
 
 fn mprotect_lazy_pages(mem: &dyn Memory, protect: bool) -> Result<(), Error> {
-    let wasm_mem_addr = mem.get_buffer_host_addr();
-    gear_ri::mprotect_lazy_pages(wasm_mem_addr, protect).map_err(Into::into)
+    let wasm_mem_addr = match mem.get_buffer_host_addr() {
+        None => return Ok(()),
+        Some(addr) => addr,
+    };
+    gear_ri::mprotect_lazy_pages(wasm_mem_addr, protect)
+        .map_err(Into::into)
+        .map_err(|e| {
+            log::error!("{} (it's better to stop node now)", e);
```

Comment:
just for clarification: this message means, that after getting error node can be up. Is that true? What if validator don't stop the node? Shall we stop it for validator?

Reply:
Yes we can but It can be unsafe - because we cannot guaranty that this won't happen for all validators, so in the case all nodes will fall down - and we do not wanna this